### PR TITLE
allow closure types

### PR DESCRIPTION
**Pull request by plicease**
This lets you use closure types which look something like this:  `(int)->int` or `(int,int)->int`.

### CODE REVIEW

**Comment by plicease** [comment]
Thanks for DC.PM (http://dc.pm.org) for working through how to get closures to work last night!

I've combined #1, #2, #3 and #4 as well as a `.gitignore` into my combined branch:
https://github.com/plicease/NativeCall/tree/combined
with the conflicts already fixed.

**Comment by mohawk2** [comment]
I do like using `/r` in code, but since this module is a bit toolchain-ish, I don't want to require 5.14. Could you rework it to not need `/r`?

**Comment by mohawk2** [comment]
Also there's no docs to explain what the "closure type" actually means!

**Comment by mohawk2** [comment]
Having read and merged #2 I'm now happy with the 5.14 and the `/r`. We still need a note in the docs to say this thing is possible, though.

**Comment by mohawk2** [comment]
Could you rebase this and help out with the conflicts that arise? A suitable `.gitignore` would be really helpful too!

**Comment by plicease** [comment]
sure will address this tomorrow.

**Comment by plicease** [comment]
rebased, gitignored and documented.

File: lib/NativeCall.pm [hunk 1]
```diff
@@ -22,11 +22,11 @@ sub _attr_parse {
     (\w+)
     (?:
       \(
-      (.*?)
+      (.*)
```
Comment (mohawk2) [inline lib/NativeCall.pm:25]: Doesn't this change make it greedy-match all the way to the last bracket? Everything else looks great, just want a bit of help understanding this.

Comment (plicease) [reply lib/NativeCall.pm:25]: Right.  It needs to be greedy in order to capture the closure specification (int)->int.  Without the greedy you get these failures with the test that I constructed:

```
crux% prove -lv t/closure.t 
t/closure.t .. 
not ok 1

#   Failed test at t/closure.t line 18.
#     Structures begin differing at:
#          $got->[0] = '(int'
#     $expected->[0] = '(int)->int'
not ok 2

#   Failed test at t/closure.t line 21.
#     Structures begin differing at:
#          $got->[0] = '(int'
#     $expected->[0] = '(int)->int'
not ok 3

#   Failed test at t/closure.t line 24.
#     Structures begin differing at:
#          $got->[0] = '(int'
#     $expected->[0] = '(int,int)->int'
not ok 4

#   Failed test at t/closure.t line 27.
#     Structures begin differing at:
#          $got->[0] = '(int'
#     $expected->[0] = '(int,int)->int'
1..4
# Looks like you failed 4 tests of 4.
Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/4 subtests 

Test Summary Report
-------------------
t/closure.t (Wstat: 1024 Tests: 4 Failed: 4)
  Failed tests:  1-4
  Non-zero exit status: 4
Files=1, Tests=4,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.04 cusr  0.01 csys =  0.09 CPU)
Result: FAIL
```

**Pull request merged**